src/main/java/com/umc/networkingService/domain/branch/service/BranchServiceImpl.java

```diff
@@ -1,19 +1,143 @@
+import com.umc.networkingService.domain.university.entity.University;
+import com.umc.networkingService.global.common.enums.Semester;
 import com.umc.networkingService.global.common.exception.ErrorCode;
-import com.umc.networkingService.global.common.exception.RestApiException;
+import com.umc.networkingService.global.utils.S3FileComponent;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
 
+import java.util.List;
+import java.util.Optional;
+import com.umc.networkingService.global.common.exception.RestApiException;
 import java.util.UUID;
 
 @Service
 @RequiredArgsConstructor
-public class BranchServiceImpl implements BranchService{
+public class BranchServiceImpl implements BranchService {
 
     private final BranchRepository branchRepository;
+    private final BranchUniversityRepository branchUniversityRepository;
+    private final S3FileComponent s3FileComponent;
+
+    private static final String BRANCH_CATEGORY = "branch";
+    private static final Integer PAGE_SIZE = 10;
+
+    @Transactional         //지부 생성
+    public UUID postBranch(BranchRequest.PostBranchDTO request) {
+
+        //이름이 없는 경우
+        if(request.getName() == null || request.getName().isEmpty()){
+            throw new BranchHandler(ErrorCode.BRANCH_NAME_EMPTY);
+        }
+        //설명이 없는 경우
+        if(request.getDescription() == null || request.getDescription().isEmpty()){
+            throw new BranchHandler(ErrorCode.BRANCH_DESCRIPTION_EMPTY);
+        }
+
+        Branch newBranch = BranchConverter
+                .toBranch
+                        (request
+                        ,uploadImageS3(BRANCH_CATEGORY,request.getImage())
+                        );
+
+        Branch savedBranch = branchRepository.save(newBranch);
+        if (savedBranch.getId() == null) {
+            throw new BranchHandler(ErrorCode.BRANCH_SAVE_FAIL);
+        }
+
+
+        return branchRepository.save(newBranch).getId();
+    }
+
+    @Transactional          //지부 수정
+    public UUID patchBranch(BranchRequest.PatchBranchDTO request) {
+
+        Optional<Branch> optionalBranch = branchRepository.findById(request.getBranchId());
+        if(optionalBranch.isEmpty()){
+            throw new BranchHandler(ErrorCode.BRANCH_NOT_FOUND);
+        }
+        //이름이 없는 경우
+        if(request.getName() == null || request.getName().isEmpty()){
+            throw new BranchHandler(ErrorCode.BRANCH_NAME_EMPTY);
+        }
+        //설명이 없는 경우
+        if(request.getDescription() == null || request.getDescription().isEmpty()){
+            throw new BranchHandler(ErrorCode.BRANCH_DESCRIPTION_EMPTY);
+        }
+
+
+
+        Branch branch = optionalBranch.get();
+        branch.updateBranch(request, uploadImageS3(BRANCH_CATEGORY, request.getImage()));
+        return branchRepository.save(branch).getId();
```

Comment:
Transactional 어노테이션을 사용하기 때문에, university 엔티티의 변경사항이 트랜잭션이 종료될 때 자동으로 DB에 반영되기 때문에 추가적인 save 메소드는 사용하지 않아도 좋을 거 같아요!!
```suggestion
        return branch.getId();
```